src/kzg10/mod.rs

```diff
@@ -536,23 +560,28 @@ mod tests {
         assert_eq!(f_comm, f_comm_2);
     }
 
-    fn end_to_end_test_template<E: PairingEngine>() -> Result<(), Error> {
+    fn end_to_end_test_template<E, P>() -> Result<(), Error>
+    where
+        E: PairingEngine,
+        P: UVPolynomial<E::Fr, Point = E::Fr>,
+        for<'a, 'b> &'a P: Div<&'b P, Output = P>,
+    {
         let rng = &mut test_rng();
         for _ in 0..100 {
             let mut degree = 0;
             while degree <= 1 {
                 degree = usize::rand(rng) % 20;
             }
-            let pp = KZG10::<E>::setup(degree, false, rng)?;
-            let (ck, vk) = KZG10::trim(&pp, degree)?;
-            let p = Polynomial::rand(degree, rng);
+            let pp = KZG10::<E, P>::setup(degree, false, rng)?;
+            let (ck, vk) = KZG10::<E, P>::trim(&pp, degree)?;
+            let p = rand_poly::<E, P>(degree, None, rng);
```

Comment:
Why does `Polynomial::rand` no longer work?

Reply:
Oh I guess for kzg templates this would be fine good point.